src/persist-client/src/batch.rs

```diff
@@ -156,6 +157,23 @@ where
         self.mark_consumed();
         ret
     }
+
+    /// Turns this [`Batch`] into a [`WriterEnrichedHollowBatch`], which can be
+    /// used to transfer this batch across process boundaries, for example when
+    /// exchanging data between timely workers.
```

Comment:
make a note in the doc about losing track of this resulting in leaked state in durable storage and that the caller is responsible for ensuring it's turned back into a Batch